runtime/hydradx/src/evm/mod.rs

```diff
@@ -142,11 +135,31 @@ impl pallet_evm::Config for crate::Runtime {
 	type FeeCalculator = crate::DynamicEvmFee;
 	type FindAuthor = FindAuthorTruncated<Aura>;
 	type GasWeightMapping = pallet_evm::FixedGasWeightMapping<Self>;
-	type OnChargeTransaction = TransferEvmFees<DealWithFees>;
+	type OnChargeTransaction = evm_fee::TransferEvmFees<
+		evm_fee::DepositEvmFeeToTreasury,
+		crate::MultiTransactionPayment, // Get account's fee payment asset
+		WethAssetId,
+		ConvertToCurrencyUsingOracle<
```

Comment:
why is this necessary? `OraclePriceAdapter` should do everything you need same as for multi currency payment, including the specification of `Period` which is `SHORT` by default, not last block. `SHORT` is used by multicurrency as well

Reply:
what about `AssetFeeOraclePriceProvider` ? :)